File: notes_server/src/main/java/com/urarik/notes_server/analysis/table/Relationship.java
package com.urarik.notes_server.analysis.table;

import javax.persistence.*;
import javax.persistence.Entity;

@Entity
@Table(name="RELATIONSHIP")
public class Relationship {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long _id;

    @Column(nullable = false, updatable = false, name = "id")
    private Long id;

    @Column
    Long fromId;
    @Column
    Long toId;
    @Column
    Long theOrder;
    @Column
    Long size;
    @Column
    String type;
    @Column
    Double height;
    @Column
    Double edgeWidth;

    public Relationship() {
    }


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getFromId() {
        return fromId;
    }

    public void setFromId(Long fromId) {
        this.fromId = fromId;
    }

    public Long getToId() {
        return toId;
    }

    public void setToId(Long toId) {
        this.toId = toId;
    }

    public Long getTheOrder() {
        return theOrder;
    }

    public void setTheOrder(Long theOrder) {
        this.theOrder = theOrder;
    }

    public Long getSize() {
        return size;
    }

    public void setSize(Long size) {
        this.size = size;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Double getHeight() {
        return height;
    }

    public void setHeight(Double height) {
        this.height = height;
    }

    public Double getEdgeWidth() {
        return edgeWidth;
    }

    public void setEdgeWidth(Double edgeWidth) {
        this.edgeWidth = edgeWidth;
    }
}

File: client/src/components/analysis_modules/classdiagram/classes/Relationship.js
export default class Relationship {
    constructor(id, fromId, toId, order, size, type, height = 2, edgeWidth = 25) {
        this.id = id;
        this.fromId = fromId;
        this.toId = toId;
        this.order = order;
        this.size = size;
        this.type = type;

        this.height = height;
        this.edgeWidth = edgeWidth;
    }

    static getInstanceFromSave({id, fromId, toId, theOrder, size, type, height, edgeWidth}) {
        return new Relationship(id, fromId, toId, theOrder, size, type, height, edgeWidth);
    }

    adjust(percentage) {
        if(percentage !== undefined)
            this.edgeWidth += this.edgeWidth * percentage;   
    }

    getProperties() {
        return [this.id, this.fromId, this.toId, this.order, this.size, this.type, this.edgeWidth];
    }
}